QuestIDTracking.lua

```diff
@@ -9,24 +9,25 @@ local hadquest
 
 local turnedinquests, currentquests, oldquests, titles, firstscan, abandoning = {}, {}, {}, {}, true
 TourGuide.turnedinquests = turnedinquests
--- local qids = setmetatable({}, {
--- 	__index = function(t,i)
--- 		local v = tonumber(i:match("|Hquest:(%d+):"))
--- 		t[i] = v
--- 		return v
--- 	end,
--- })
--- TourGuide.QIDmemo = qids
+--[[ local qids = setmetatable({}, {
+	__index = function(t,i)
+		local v = tonumber(i:match("|Hquest:(%d+):"))
+		t[i] = v
+		return v
+	end,
+})
+TourGuide.QIDmemo = qids ]]
 
 
 function TourGuide:QuestID_QUEST_LOG_UPDATE()
+	local numEntries, numQuests = GetNumQuestLogEntries();
+
 	currentquests, oldquests = oldquests, currentquests
 	for i in pairs(currentquests) do currentquests[i] = nil end
 
-	for i=1,GetNumQuestLogEntries() do
-		local title = GetQuestLogTitle(i)
-		local qid = select(8, title) or 0
-
+	for i=1, numEntries do
+		local title, _, _, _, _, _, _, qid = GetQuestLogTitle(i)
+		qid = qid or 0
 		if qid > 0 then
 			currentquests[qid] = true
 			titles[qid] = title
```

Comment:
I think in the end of this, we should put some kind of check that raises a warning if the number of quests in currentquests / titles is not the same as numQuests.

Reply:
In general I just think it's much faster to read and understand what API returns. In the original, you kinda have to read an extra line to get the same information.

This is especially true when those API names are quite terrible, you could think that it returns just a single string, but it actually returns quite a lot more.

Reply:
As an example, Blizzard tries to avoid the use of select in this particular manner for good reason. In their UI code, when they use it, they actually wrap it within a function to make it more readable/understandable:

```Lua
local GetFaction = function()
    return select(1, UnitFactionGroup(PTR_IssueReporter.Data.UnitTokens.Player))
end
local GetRaceID = function()
    return select(3, UnitRace(PTR_IssueReporter.Data.UnitTokens.Player))
end
```